codalab/worker/worker_run_state.py

```diff
@@ -457,15 +506,20 @@ def _transition_from_CLEANING_UP(self, run_state):
                     time.sleep(1)
 
         for dep in run_state.bundle.dependencies:
-            dep_key = DependencyKey(dep.parent_uuid, dep.parent_path)
             if not self.shared_file_system:  # No dependencies if shared fs worker
+                dep_key = DependencyKey(dep.parent_uuid, dep.parent_path)
                 self.dependency_manager.release(run_state.bundle.uuid, dep_key)
 
-            child_path = os.path.join(run_state.bundle_path, dep.child_path)
-            try:
-                remove_path(child_path)
-            except Exception:
-                logger.error(traceback.format_exc())
+            if dep.child_path == RunStateMachine._CURRENT_DIRECTORY:
```

Comment:
Could we avoid having this logic duplicated here by having the first time when we are mounting things produce a list of things to remove (like `to_mount`)?  Then, we just need to iterate over that list here.

Reply:
Yes, that was much cleaner. I updated this PR.